Controllers/GenericController.cs

```diff
@@ -209,10 +209,10 @@ public dynamic GetJob(string jobId)
                         }
                     }
 
-					if (method.Parameters.Any(param => queryStrings.All(q => q.Key != param.Name)))
+					if (method.Parameters.Any(param => queryStrings.All(q => q.Key != param.Name))) 
 					{
 						DynamicPowershellApiEvents.Raise.VerboseMessaging(String.Format("Cannot find all parameters required."));
-						throw new MissingParametersException("Cannot find all parameters required.");
```

Comment:
Maybe check if the missing parameter is optional (`!Required`) and only throw if it's not?